src/components/shared/Activity/index.tsx

```diff
@@ -1,108 +1,366 @@
+              ]}
+            >
+              {nextInChain &&
+                [
+                  ActivityDirection.Bidirectional
+                  // ActivityDirection.Backward
+                ].includes(nextInChain.direction) && (
+                  <ArrowHead
+                    direction={ArrowHeadDirection.Up}
+                    width={12}
+                    color={palette.veryLightGray}
+                  />
+                )}
+              <View style={styles.chainIndicatorLine} />
+              {nextInChain &&
+                [
+                  ActivityDirection.Bidirectional,
+                  ActivityDirection.Forward
+                ].includes(nextInChain.direction) && (
+                  <ArrowHead
+                    direction={ArrowHeadDirection.Down}
+                    width={12}
+                    color={palette.veryLightGray}
+                  />
+                )}
+            </View>
+            <ActivityContentBody
+              body={body}
+              onFindVideoElems={elems =>
+                this.props.onFindVideoElems([...this.ownVideoElems, ...elems])
+              }
+            />
+          </View>
+        )}
+        {nextInChain && (
+          <ActivityContent
+            content={nextInChain.content}
+            onFindVideoElems={elems =>
+              this.props.onFindVideoElems([...this.ownVideoElems, ...elems])
+            }
+          />
+        )}
+      </View>
+    );
+  }
 }
 
-/**
- * Addendum to any Activity that invites a member to take an action.
- */
-export interface ActivityCallToAction {
-  actor: Member;
-  actions: CallToAction[];
-}
+export class ActivityView extends React.Component<
+  Props & NavigationInjectedProps,
+  {}
+> {
+  videoElems?: VideoWithPlaceholderView[];
 
-/**
- * A full description of any conceptually whole activity that happens on Raha.
- *
- * id just needs to be unique and deterministically derived from the content.
- */
-export interface Activity extends ActivityContent {
-  id: string;
-  timestamp: Date;
-  content: ActivityContent;
-  callToAction?: ActivityCallToAction;
+  /**
+   * Reset video playback state, stop it
+   */
+  public resetVideo = () => {
+    if (this.videoElems) {
+      for (const elem of this.videoElems) {
+        elem.reset();
+      }
+    }
+  };
+
+  render() {
+    const { content, callToAction, timestamp } = this.props.activity;
+    return (
+      <View style={styles.activity}>
+        <ActivityContent
+          content={content}
+          onFindVideoElems={elems => {
+            this.videoElems = elems;
+          }}
+        />
+        {callToAction && <CallToAction callToAction={callToAction} />}
+        <View style={metadataRowStyle}>
+          <Text style={styles.timestamp}>
+            {formatRelative(timestamp, new Date()).toUpperCase()}
+          </Text>
+        </View>
+      </View>
+    );
+  }
 }
+
+const leftColumnWidth = 50;
```

Comment:
When there's this much style info, how do you feel about a `index.styles.ts` file or something that we import from?

Reply:
That seems fine by me but not really that important